### PR TITLE
Adds Husky hooks for linting.

### PR DESCRIPTION
**Pull request by mmcachran**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -187,5 +189,18 @@
...
+    "hooks": {
+      "pre-commit": [
+        "lint-staged"
+      ]
+    }
+  },
+  "lint-staged": {
+    "*.php": [
+      "php -l",
+      "./vendor/bin/phpcs --extensions=php --warning-severity=8 -s"
```
Comment (jbrinley) [inline package.json:203]: Can this use our phpcs config? This is how we run the tests in GH actions: 

```
./vendor/bin/phpcs --standard=./phpcs.xml
```

**Review by jbrinley — COMMENTED** [review]
Great idea, @mmcachran

**Comment by defunctl** [comment]
@mmcachran : Definitely like where this idea is going! We'll still need a changelog.md update. 

Can this also hook into `.git/hooks/prepare-commit-msg`? 

I'd love to see ticket prefixing for commits, e.g. `[PRO-111] fixed some thing`, based on the branch structure of `feature/<ticket_id>/description`.

**Comment by mmcachran** [comment]
> @mmcachran : Definitely like where this idea is going! We'll still need a changelog.md update.
> 
> Can this also hook into `.git/hooks/prepare-commit-msg`?
> 
> I'd love to see ticket prefixing for commits, e.g. `[PRO-111] fixed some thing`, based on the branch structure of `feature/<ticket_id>/description`.

This is possible, but since we don't track the .git directory in this repo it'll need to be changed directly after cloning the project or as part of the SQ1 create script. Attached is an example of how we can do this on the FoodChainID project.

[prepare-commit-msg.zip](https://github.com/moderntribe/square-one/files/5928479/prepare-commit-msg.zip)

**Comment by mmcachran** [comment]
Closed in favor of https://github.com/moderntribe/square-one/pull/676

**Pull request closed without merging**